### DISCUSSION TITLE
Functions shared between custom pages

### DISCUSSION THREAD

**Discussion by Iainreynard** [filamentphp/filament#11727] [Help] [answered]
### Package

Other

### Package Version

v3

### How can we help you?

I have a function that I want to call on various custom pages. In the traditional Laravel framework, I would have put this in a service, then added the __construct() notation to the Controller which needs the function. However, I dont seem to be able to do this in Filament. 

Details:

Here is the function test(), written into a Service called tripService:

`<?php

namespace App\Services;

class TripService

{

    public function test(){

        dd('yes');

    }

}`

Then, I use the construct() function in the top of my custom page called LocationVolumesView:

`class LocationVolumeView extends Page implements HasForms, HasTable
{

    private $tripService;

   public function __construct (TripService $tripService)
   {
       $this->tripService = $tripService;
   }`

However, I get the following error:

Too few arguments to function App\Filament\Pages\LocationVolumeView::__construct(), 0 passed in ~/vendor/livewire/livewire/src/Features/SupportPageComponents/SupportPageComponents.php on line 212 and exactly 1 expected

Any help appreciated

**Comment by leandrocfe** [comment, marked as answer]
Pages are [Livewire Components](https://livewire.laravel.com/docs/components)

You could create a trait and use it in all pages

**Reply by Iainreynard** [reply]
Fantastic - works perfectly - many thanks!
